src/lib.rs

```diff
@@ -338,29 +342,35 @@ pub trait Miniconf {
         let meta = Self::metadata();
 
         if TS < meta.max_length {
-            return Err(IterError::InsufficientTopicLength);
+            return Err(IterError::PathLength);
         }
 
         if L < meta.max_depth {
-            return Err(IterError::InsufficientStateDepth);
+            return Err(IterError::PathDepth);
         }
 
-        Ok(Self::unchecked_iter_paths())
+        Ok(Self::unchecked_iter_paths(Some(meta.count)))
     }
 
     /// Create an iterator of all possible paths.
     ///
     /// This is a depth-first walk.
+    /// It will return all paths, even those that may be absent at run-time.
     ///
     /// # Note
     /// This does not check that the path size or state vector are large enough. If they are not,
     /// panics may be generated internally by the library.
     ///
+    /// # Args
+    /// * count: Optional iterator length
+    ///
     /// # Template Arguments
     /// * `L`  - The maximum depth of the path, i.e. number of separators plus 1.
     /// * `TS` - The maximum length of the path in bytes.
-    fn unchecked_iter_paths<const L: usize, const TS: usize>() -> iter::MiniconfIter<Self, L, TS> {
-        iter::MiniconfIter::default()
+    fn unchecked_iter_paths<const L: usize, const TS: usize>(
```

Comment:
Also needs a changelog entry, as this is technically (another) breaking API change